File: MultiLanguageSandbox/src/test/deps/pascal/6.pas
{$ASSERTIONS ON}
{
  Calculates the factorial of a given number N modulo 10007.
  
  Parameters:
  - N (integer): An integer representing the input value (N <= 10000).
  
  Returns:
  - integer: The factorial of N modulo 10007.
  
  Examples:
  - process_request(1) returns 1
  - process_request(2) returns 2
  - process_request(3) returns 6
}

function process_request(n: integer): integer;
var
  i, result: integer;
begin
  if n < 0 then
    process_request := 0  // Factorial is not defined for negative numbers
  else
  begin
    result := 1;
    for i := 1 to n do
    begin
      result := (result * i) mod 10007;
    end;
    process_request := result;
  end;
end;
procedure testProcessRequest;
begin
  Assert(process_request(0) = 1, 'Test for N = 0 failed');
  Assert(process_request(1) = 1, 'Test for N = 1 failed');
  Assert(process_request(2) = 2, 'Test for N = 2 failed');
  Assert(process_request(3) = 6, 'Test for N = 3 failed');
  Assert(process_request(4) = 24, 'Test for N = 4 failed');
  Assert(process_request(10) = 6266, 'Test for N = 10 failed');
  Assert(process_request(10000) = 6991, 'Test for N = 10000 failed');
end;

begin
  (* Run the tests *)
  testProcessRequest;
  // WriteLn('All tests passed.');
end.

File: MultiLanguageSandbox/src/test/deps/lua/6.lua

--[[
    Find the factorial of N and take the modulo 10007 of the result.
​    
    Parameters:
    - N: An integer representing the input value (N <= 10000).
    
    Returns:
    - The result after taking the modulo 10007 of the output.

    Examples:
    assert(process_request(1) == 1)
    assert(process_request(5) == 10)
]]

function process_request(n)
    if n < 0 then
        return 0 -- Factorial is not defined for negative numbers
    end
    
    local result = 1
    for i = 1, n do
        result = (result * i) % 10007
    end
    return result
end

-- Test cases
assert(process_request(1) == 1)
assert(process_request(5) == 120)
local function test_process_request()
    assert(process_request(0) == 1) -- Added test for boundary condition
    assert(process_request(1) == 1)
    assert(process_request(2) == 2)
    assert(process_request(3) == 6)
    assert(process_request(4) == 24)
    assert(process_request(10) == 6266) -- 10! % 10007 = 3628800 % 10007 = 6266
    assert(process_request(10000) == 6991) -- Added a test for upper boundary condition
end

test_process_request()
-- No equivalent in Lua for printf statement to indicate test status, as assert will automatically throw an error when a test fails.